pdns/dnsdistdist/dnsdist-metrics.cc

```diff
@@ -176,18 +178,16 @@ std::optional<std::string> declareCustomMetric(const std::string& name, const st
   const std::string finalCustomName(customName ? *customName : "");
   if (type == "counter") {
     auto customCounters = s_customCounters.write_lock();
-    auto itp = customCounters->insert({name, MutableCounter()});
+    auto itp = customCounters->emplace(name, std::map<std::string, MutableCounter>());
     if (itp.second) {
-      g_stats.entries.write_lock()->emplace_back(Stats::EntryPair{name, &(*customCounters)[name].d_value});
```

Comment:
One problem with this change is that a custom metric that is never set/incremented/decremented doesn't show up in the metrics anymore. That breaks our regression tests, but more importantly I think it breaks some user expectations. Do you think it would make sense to add an optional parameter (perhaps we should make it an optional LuaAssociativeTable right away so we don't end up with too many parameters) to this method to specify whether the metrics is expected to have labels? Metrics without labels (the default since it is the existing behaviour) would be added to `g_stats` right away, and metrics with labels wouldn't.

Reply:
Alright, that makes sense. I just removed it since it caused issues for ones with labels, but I didn't expect to break the regular use case :smile: 
I assume we can make the old behavior default and then only skip this step if explicitly marked as a metric with labels.

I agree about the optional LuaAssociativeTable, since all of the other metrics related functions have these now.